.github/workflows/format-check.yml

```diff
@@ -20,3 +20,20 @@ jobs:
         run: |
             pip3 install --upgrade autopep8
             ./scripts/run-autopep8.sh
+
+  clippy-check:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout Kani
+        uses: actions/checkout@v2
+
+      - name: Setup Kani Dependencies
+        uses: ./.github/actions/setup
+        with:
+            os: ubuntu-20.04
+
+      - name: "Run Clippy"
+        run: |
+          cargo clippy --all -- -D warnings \
+          -A "clippy::expect_fun_call" \
+          -A "clippy::or_fun_call"
```

Comment:
Can this specific configuration be put in a clippy.toml or something similar?
If I run cargo clippy on my machine, I want it to run just like on the CI, since this is a repo-wide decision, not just a CI decision.

And then, they can also be removed from the lib files

Reply:
@giltho I think that would be good as well. I'm less sure of the exact method to implement it. If you know of a working example, please send me the link.

A quick search suggests the method for doing that seems to be through `.cargo/config.toml`. https://github.com/emilk/egui/blob/e76c919c7e70c208c9a6209b9fe3369e7b6db99d/.cargo/config.toml

Let me prototype a little bit b/c I am not sure if this will have negative effects. I read **unconfirmed** rumors that this negatively effects incremental recompile.

Reply:
Update:
1. Incremental recompile issue can be mitigated.
2. However, .cargo/config.toml does not play nicely with the custom build command for crates under `library`.
```
error: failed to run custom build command for `kani-compiler v0.6.0 (/Users/ytakashl/Desktop/kani/kani-compiler)`

Caused by:
  process didn't exit successfully: `/Users/ytakashl/Desktop/kani/target/debug/build/kani-compiler-bfa3790a81ae900c/build-script-build` (exit status: 1)
  --- stdout
  cargo:rustc-link-arg-bin=kani-compiler=-Wl,-rpath,/Users/ytakashl/.rustup/toolchains/nightly-2022-07-19-aarch64-apple-darwin/lib
  cargo:rustc-link-arg-bin=kani-compiler=-Wl,-rpath,@loader_path/../toolchain/lib
  cargo:rerun-if-changed=../library/kani

  --- stderr
     Compiling kani_macros v0.6.0 (/Users/ytakashl/Desktop/kani/library/kani_macros)
  error: length comparison to zero
    --> library/kani_macros/src/lib.rs:42:13
     |
  42 |     assert!(attr.to_string().len() == 0, "#[kani::proof] does not take any arguments");
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: using `is_empty` is clearer and more explicit: `attr.to_string().is_empty()`
     |
     = note: `-D clippy::len-zero` implied by `-D warnings`
```

While there are only 4 warnings under `library/`, one of the warnings is a bit thorny b/c it asks us to define a safety doc comment for Invariant.

Reply:
Resolved by adding `.cargo/config.toml`.

Warnings under `library/` are fixed since they appear not to be ignored.